ballerina-tests/tests/wso2-model-provider-tests.bal

```diff
@@ -225,24 +225,50 @@ type Cricketers record {|
     string name;
 |};
 
+@test:Config
+function testGenerateMethodWithStringUnionNull() returns error? {
+    string? result = check defaultModelProvider->generate(`Give me a random joke`);
+    test:assertTrue(result is string);
+}
+
+@test:Config
+function testGenerateMethodWithRecUnionBasicType() returns error? {
+    Cricketers|string result = check defaultModelProvider->generate(`Give me a random joke about cricketers`);
+    test:assertTrue(result is string);
+}
+
+@test:Config
+function testGenerateMethodWithRecUnionNull() returns error? {
+    Cricketers? result = check defaultModelProvider->generate(`Name a random world class cricketer in India`);
+    test:assertTrue(result is Cricketers);
+}
+
+@test:Config
+function testGenerateMethodWithArrayOnly() returns error? {
+    Cricketers[] result = check defaultModelProvider->generate(`Name 10 world class cricketers in India`);
```

Comment:
When we reuse the same type we aren't properly testing the compile-time behaviour, right? Are they covered in other tests?

Reply:
My point is that if we have both

```
T? x = check ....;

T[] y = check ....;
```

the schema will be added to `T` at compile-time even if we remove the logic for one case. To properly test the different paths we need different types.